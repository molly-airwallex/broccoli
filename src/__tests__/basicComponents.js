import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import injectTapEventPlugin from 'react-tap-event-plugin';


import Header from '../scripts/components/header';
import Footer from '../scripts/components/footer';
import SignupForm from '../scripts/components/signupForm';
import Signup from '../scripts/components/signup';


injectTapEventPlugin();
const store = createStore(() => ({
  broccoliApp : {
    signup: {
      signedUp: false,
      signingUp: false,
      pending: false,
      error: false
    }
  }
}));

test('Header component', () => {
  const component = renderer.create(
    <MuiThemeProvider>
      <Header />
    </MuiThemeProvider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Footer component', () => {
  const component = renderer.create(
    <Footer />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('SignupForm component', () => {
  const component = renderer.create(
    <Provider store={store}>
      <MuiThemeProvider>
        <SignupForm />
      </MuiThemeProvider>
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Signup component', () => {
  const component = renderer.create(
      <MuiThemeProvider>
        <Signup />
      </MuiThemeProvider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});