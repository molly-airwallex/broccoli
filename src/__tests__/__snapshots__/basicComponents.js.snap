// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component 1`] = `
<div
  className="footer"
>
  <div>
    Made with 
    <span
      aria-hidden={true}
      className="fa fa-heart"
      style={
        Object {
          "color": "red",
        }
      }
    />
     in Melbourne.
  </div>
  <div>
    <span
      aria-hidden={true}
      className="fa fa-copyright"
    />
     Broccoli & Co. All rights reserved.
  </div>
</div>
`;

exports[`Header component 1`] = `
<div
  className={undefined}
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#4dc581",
      "borderRadius": "0px",
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "display": "flex",
      "fontFamily": "Roboto, sans-serif",
      "height": 64,
      "minHeight": 64,
      "muiPrepared": true,
      "paddingLeft": "15%",
      "paddingRight": "15%",
      "position": "relative",
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "width": "100%",
      "zIndex": 1100,
    }
  }
>
  <h1
    onTouchTap={[Function]}
    style={
      Object {
        "boxFlex": 1,
        "color": "#ffffff",
        "flex": "1",
        "fontSize": 24,
        "fontWeight": 400,
        "height": 64,
        "letterSpacing": 0,
        "lineHeight": "64px",
        "margin": 0,
        "muiPrepared": true,
        "overflow": "hidden",
        "paddingTop": 0,
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
      }
    }
  >
    BROCCOLI & CO.
  </h1>
</div>
`;

exports[`Signup component 1`] = `
<div
  className="Aligner"
>
  <div
    className="title"
  >
    A better way
  </div>
  <div
    className="title"
  >
    to enjoy every  day.
  </div>
  <div
    className="subtitle"
  >
    Be the first to know
  </div>
  <div
    className={undefined}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#ffffff",
        "borderRadius": 2,
        "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
               0 1px 4px rgba(0, 0, 0, 0.12)",
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "minWidth": 88,
        "muiPrepared": true,
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      }
    }
  >
    <button
      disabled={false}
      href={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      onTouchTap={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "#4dc581",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": "100%",
          "zIndex": 1,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <div
          style={
            Object {
              "backgroundColor": false,
              "borderRadius": 2,
              "height": 36,
              "muiPrepared": true,
              "top": 0,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          <span
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "margin": 0,
                "muiPrepared": true,
                "opacity": 1,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "userSelect": "none",
              }
            }
          >
            Request an invite
          </span>
        </div>
      </div>
    </button>
  </div>
</div>
`;

exports[`SignupForm component 1`] = `
<div>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="title"
    >
      Request an invite
    </div>
    <div
      className="hr"
    />
    <div>
      <div
        className={undefined}
        style={
          Object {
            "backgroundColor": "transparent",
            "border": "1px solid black",
            "cursor": "auto",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "35px",
            "height": 48,
            "lineHeight": "24px",
            "margin": "8px",
            "muiPrepared": true,
            "paddingLeft": "8px",
            "paddingRight": "8px",
            "position": "relative",
            "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "90%",
          }
        }
      >
        <div
          style={
            Object {
              "bottom": 12,
              "color": "rgba(0, 0, 0, 0.3)",
              "muiPrepared": true,
              "opacity": 1,
              "position": "absolute",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          Full name
        </div>
        <input
          disabled={false}
          id="name"
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "appearance": "textfield",
              "backgroundColor": "rgba(0,0,0,0)",
              "border": "none",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "inherit",
              "font": "inherit",
              "fontSize": "40px",
              "height": "100%",
              "lineHeight": "40px",
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "width": "100%",
            }
          }
          type="text"
          value=""
        />
      </div>
    </div>
    <div>
      <div
        className={undefined}
        style={
          Object {
            "backgroundColor": "transparent",
            "border": "1px solid black",
            "cursor": "auto",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "35px",
            "height": 48,
            "lineHeight": "24px",
            "margin": "8px",
            "muiPrepared": true,
            "paddingLeft": "8px",
            "paddingRight": "8px",
            "position": "relative",
            "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "90%",
          }
        }
      >
        <div
          style={
            Object {
              "bottom": 12,
              "color": "rgba(0, 0, 0, 0.3)",
              "muiPrepared": true,
              "opacity": 1,
              "position": "absolute",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          Email
        </div>
        <input
          disabled={false}
          id="email"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "appearance": "textfield",
              "backgroundColor": "rgba(0,0,0,0)",
              "border": "none",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "inherit",
              "font": "inherit",
              "fontSize": "40px",
              "height": "100%",
              "lineHeight": "40px",
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "width": "100%",
            }
          }
          type="text"
          value=""
        />
      </div>
    </div>
    <div>
      <div
        className={undefined}
        style={
          Object {
            "backgroundColor": "transparent",
            "border": "1px solid black",
            "cursor": "auto",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "35px",
            "height": 48,
            "lineHeight": "24px",
            "margin": "8px",
            "muiPrepared": true,
            "paddingLeft": "8px",
            "paddingRight": "8px",
            "position": "relative",
            "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "90%",
          }
        }
      >
        <div
          style={
            Object {
              "bottom": 12,
              "color": "rgba(0, 0, 0, 0.3)",
              "muiPrepared": true,
              "opacity": 1,
              "position": "absolute",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          Confirm Email
        </div>
        <input
          disabled={false}
          id="confirmEmail"
          name="confirmEmail"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "appearance": "textfield",
              "backgroundColor": "rgba(0,0,0,0)",
              "border": "none",
              "color": "rgba(0, 0, 0, 0.87)",
              "cursor": "inherit",
              "font": "inherit",
              "fontSize": "40px",
              "height": "100%",
              "lineHeight": "40px",
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "width": "100%",
            }
          }
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className={undefined}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "boxShadow": undefined,
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "height": "40px",
          "lineHeight": "40px",
          "marginTop": "40px",
          "minWidth": 88,
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": "90%",
        }
      }
    >
      <button
        disabled={true}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgb(229, 229, 229)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "cursor": "default",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": "40px",
            "lineHeight": "40pxpx",
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": "100%",
            "zIndex": 1,
          }
        }
        tabIndex={-1}
        type="submit"
      >
        <div
          style={
            Object {
              "backgroundColor": false,
              "borderRadius": 2,
              "height": "40px",
              "muiPrepared": true,
              "top": 0,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          <span
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.3)",
                "fontSize": 36,
                "fontWeight": 500,
                "height": "40px",
                "letterSpacing": 0,
                "lineHeight": "40px",
                "margin": 0,
                "muiPrepared": true,
                "opacity": 1,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "userSelect": "none",
              }
            }
          >
            Send
          </span>
        </div>
      </button>
    </div>
  </form>
</div>
`;
